test/gen_casadi_test.py

```diff
@@ -660,20 +662,22 @@ def test_delay_for_loop_with_expand_vectors(self):
         def _array_mx(name, n):
             return np.array([ca.MX.sym("{}[{}]".format(name, i+1)) for i in range(n)])
 
-        x = _array_mx("x", 2)
-        y = _array_mx("y", 2)
-        a = _array_mx("a", 2)
-        z = _array_mx("z", 2)
+        x = _array_mx("x", 3)
+        y = _array_mx("y", 3)
+        a = _array_mx("a", 3)
+        z = _array_mx("z", 3)
         at3_delayed = _array_mx("_pymoca_delay_0", 2)
         delay_time = ca.MX.sym("delay_time")
+        eps = _array_mx("eps", 1)
 
         ref_model.alg_states = list(map(Variable, [*x, *y, *a]))
+        ref_model.parameters = list(map(Variable, [*eps]))
         ref_model.inputs = list(map(Variable, [*at3_delayed, *z, delay_time]))
         ref_model.inputs[-1].fixed = True
-        # TODO: "a" is not yet deteced as an alias of "x" when expanding.
-        ref_model.equations = [*(x - 5 * z), *(y - at3_delayed), *(a - x)]
+        # TODO: "a" is not yet detected as an alias of "x" when expanding.
+        ref_model.equations = [*(x[1:3] - 5 * z[1:3] * eps), *(y[1:3] - at3_delayed), *(a - x)]
```

Comment:
Shouldn't this be x[2:3], z[2:3] and y[2:3] since the model now states "for i in 2:3 loop"?

Reply:
Modelica is 1-based, and Python is 0-based. Furthermore, Python uses exclusive indexing for the end, whereas Modelica uses inclusive indexing.